examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -112,83 +117,86 @@ def load_mtpl2(n_samples=100000):
+       regressor__sample_weight=df_train["Exposure"])
+
+
+##############################################################################
+# Let's compute the performance of this constant prediction baseline with 3
+# different regression metrics:
+
+from sklearn.metrics import mean_squared_error
+from sklearn.metrics import mean_absolute_error
+from sklearn.metrics import mean_poisson_deviance
 
 
 def score_estimator(estimator, df_test):
     """Score an estimator on the test set."""
-
     y_pred = estimator.predict(df_test)
 
     print("MSE: %.3f" %
           mean_squared_error(df_test["Frequency"], y_pred,
-                             df_test["Exposure"]))
+                             sample_weight=df_test["Exposure"]))
     print("MAE: %.3f" %
           mean_absolute_error(df_test["Frequency"], y_pred,
-                              df_test["Exposure"]))
+                              sample_weight=df_test["Exposure"]))
 
     # ignore non-positive predictions, as they are invalid for
     # the Poisson deviance
     mask = y_pred > 0
     if (~mask).any():
-        warnings.warn("Estimator yields non-positive predictions for {} "
-                      "samples out of {}. These will be ignored while "
-                      "computing the Poisson deviance"
-                      .format((~mask).sum(), mask.shape[0]))
+        n_clipped, n_samples = (~mask).sum(), mask.shape[0]
+        print(f"WARNING: Estimator yields non-positive predictions for "
+              f"{n_clipped} samples out of {n_samples}. These predictions "
+              f"will be clipped while computing the Poisson deviance.")
 
     print("mean Poisson deviance: %.3f" %
-          mean_poisson_deviance(df_test["Frequency"][mask],
-                                y_pred[mask],
-                                df_test["Exposure"][mask]))
+          mean_poisson_deviance(df_test["Frequency"], y_pred.clip(min=1e-6),
+                                sample_weight=df_test["Exposure"]))
 
 
 print("Constant mean frequency evaluation:")
 score_estimator(dummy, df_test)
 
 ##############################################################################
-# We start by modeling the target variable with the least squares linear
-# regression model,
+# (Generalized) Linear models
+# ---------------------------
+#
+# We start by modeling the target variable with the (l2 penalized) least
+# squares linear regression model, more comonly known as Ridge regression. We
+# use a low penalization as we expect such a linear model to under-fit on such
+# a large dataset.
+
+from sklearn.linear_model import Ridge
 
-ridge = make_pipeline(linear_model_preprocessor, Ridge(alpha=1.0))
-ridge.fit(df_train, df_train["Frequency"],
-          ridge__sample_weight=df_train["Exposure"])
+
+ridge_glm = Pipeline([
+    ("preprocessor", linear_model_preprocessor),
+    ("regressor", Ridge(alpha=1e-6)),
+]).fit(df_train, df_train["Frequency"],
+       regressor__sample_weight=df_train["Exposure"])
```

Comment:
I cannot comment on the lines below but the estimators aren't properly linked. Should be 

```
 :class:`~sklearn.linear_model.Ridge`
```
 and 

```
 :class:`~sklearn.compose.TransformedTargetRegressor`
```